q2_types/feature_data/_format.py

```diff
@@ -138,6 +138,7 @@ class DNAFASTAFormat(model.TextFileFormat):
     def _validate_lines(self, max_lines):
         FASTADNAValidator = re.compile(r'[ACGTURYKMSWBDHVN]+\r?\n?')
         last_line_was_ID = False
+        ids = set()
```

Comment:
> Keep track of the lines all the ids occur on so when and if you find a duplicate you can report the lines each of the duplicates came on not just the line the second one occurred on (this one seems like a good idea)

You could change this to a `dict`, the key could be the ID, the value could be the line number first seen on.

Reply:
Yeah, that's what I was thinking